src/rust/lib/frp/src/nodes.rs

```diff
@@ -1653,6 +1672,46 @@ where T1:EventOutput, T2:EventOutput, T3:EventOutput, T4:EventOutput {
 
 
 
+// ==============
+// === Filter ===
+// ==============
+
+pub struct FilterData  <T,P> { _src:T, predicate:P }
```

Comment:
the `_src` field is not needed. Please remove it here and in `Map` and check if it works (it should work). This was an obsolete design and I'm wondering how on earth it's still there! We are sometimes using something like `src2:watch::Ref<T2>` to indicate on which inputs values should be cached, but here you dont need it. For example, if you have `All4`, and when you have incoming event on 1st input, you want to sample all other inputs, thus you need to cache them. Here you'll have a simple streeam, so you dont need to remmebr any info about the input.

Reply:
I think we need the type parameter `T` in the struct to implement `HasOutput`, but but there might be another way. The `_src` field was probably included because the compiler would otherwise complain that `T` is unused. It wouldn't be able to infer lifetime constraints for `T`. I will change the definition to
```pub struct FilterData  <T,P> { phantom:PhantomData<T>, predicate:P }```
This is equivalent to the original definition without wasting memory on `_src`.

Reply:
Maybe someone should check if the lifetime constraints that we get here really make sense.